apis/v1beta1/httproute_types.go

```diff
@@ -244,6 +246,37 @@ type HTTPRouteRule struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Core
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+type HTTPRouteTimeouts struct {
+	// Request specifies an end-to-end timeout for client HTTP requests,
+	// disabled by default.
+	//
+	// Request timeout values are formatted as 1h/1m/1s/1ms and MUST BE >= 1ms.
+	//
+	// For example, the following rule will wait no longer than 2 secs for the
+	// client request to complete:
+	//
+	// ```
+	// rules:
+	//	- timeouts:
+	//	    request: 2s
+	//	  backendRefs:
+	//	    ...
+	// ```
+	//
+	// Support: Core
+	//
+	// +optional
+	Request *time.Duration `json:"request,omitempty"`
```

Comment:
Maybe something like this would work? I wonder if we would also need to override marshal/unmarshal if we want the API to return *time.Duration like this?

```suggestion
	// +kubebuilder:validation:Type=string
	// +kubebuilder:validation:Format=duration
	Request *time.Duration `json:"request,omitempty"`
```

Reply:
if we use `metav1.Duration` it seems marshal/unmarshal is already taken care of according to https://github.com/kubernetes/apimachinery/blob/0ef2ec80f7a200acb85cc11636ab3a14a765b6f8/pkg/apis/meta/v1/duration.go